### DISCUSSION TITLE
Regularization images

### DISCUSSION THREAD

**Discussion by elismasilva** [bghira/SimpleTuner#340] [General]
If add a second bucket of images with repeats setted it could work like regularization images?

**Comment by bghira** [comment]
yep, similar to onetrainer in that way.

**Reply by bghira** [reply]
additionally, `ignore_epochs = True` will allow the trainer to pretend that one dataset doesn't matter for determining when an epoch is finished. it will not reset the dataset, but it will keep sampling where it left off on the next epoch, until it is exhausted. set repeats high and ignore_epochs = True and it will constantly be adding those images in.

**Reply by elismasilva** [reply]
so if i have 16 images on main dataset to train and 2300 reg images i dont need do 16 / 2300 to put it value in repeats ? how many could be my repeat in this case with ignore_epochs.?

**Reply by elismasilva** [reply]
adding how do I force it to re-cache the buckets without always having to delete the json?

**Reply by bghira** [reply]
is there a _reason_ you have to recache the buckets? they should just remain static - right? if you add more images they should be picked up automatically

for repeats it's probably enough to have a value of 2 when there's that big of a difference in the image counts.

with just 16 images on batch size 4 you'll hit 1 epoch every 4 steps unless you have repeats enabled on that dataset too.

at the end of each of those epochs you'll still have the 2300 samples being chipped-away at. so, at the end of epoch 1, you'll have pulled maybe 8-12 images from the regularisation set.

this means you have something like 500 epochs of regularisation images when `ignore_epochs=True` on the large regularisation images dataset.

**Reply by bghira** [reply]
you can also set `probability` lower on your training set, which will sample regularisation images more often.

depends on the desired outcome and how your initial results go

**Reply by elismasilva** [reply]
so 1 epoch is total_images / batch_size
